rules/falco_rules.yaml

```diff
@@ -3420,3 +3420,38 @@
 # Application rules have moved to application_rules.yaml. Please look
 # there if you want to enable them by adding to
 # falco_rules.local.yaml.
+
+
+- macro: rocketmq_check_netty_server
+  condition: (proc.name contains NettyServer and proc.env contains ROCKETMQ)
+
+- macro: rocketmq_check_payload_for_conf_update
+  condition: (evt.type=read and evt.buffer contains 'code":25' and evt.buffer contains serializeTypeCurrentRPC and evt.buffer contains JSON)
+
+- rule: Apache Rocketmq CVE-2023-33246 Exploit Detection
+  desc: >
+    CVE-2023-33246 is exploited at the function used to update configuration & can be detected by looking at the buffer for broker code-25 and serializeTypeCurrentRPC function.
+  condition: >
+    rocketmq_check_netty_server  and rocketmq_check_payload_for_conf_update
+  output: exploit detected CVE-2023-33246 (event=%evt.type server_ip=%fd.sip server_port=%fd.sport proto=%fd.l4proto parent=%proc.pname process=%proc.name command=%proc.cmdline container_id=%container.id)
+  priority: CRITICAL
+  tags: [host, CVE_2023_33246, rocketmq, apache,Mitre_Initial_Access, Mitre_Execution,TA0001, TA0002, T1133, T1059]
+
+
+- macro: rocketmq_check_env_variable
+  condition: (proc.env contains ROCKETMQ)
+
+- macro: check_file_openat
+  condition: (fd.type=file and evt.type=openat)
```

Comment:
Could we use an existing macro? https://github.com/falcosecurity/rules/blob/main/rules/falco_rules.yaml#L36C1-L43

Reply:
Thanks for pointing this out, I shall reuse existing macros. And it is working!